app/models/piece.rb

```diff
@@ -52,72 +52,74 @@ def own_piece?(row_dest, col_dest)
-      if col_pos < col_dest
+    # check each row/col combo 1 by 1
+    # row and col both increase
+    if row_pos < row_dest && col_pos < col_dest
+      row_pos += 1
+      (row_pos...row_dest).each do |row_num|
         col_pos += 1
-        col_dest -= 1
-        return !game.pieces.find_by(captured: false, row_position: row_pos, col_position: [col_pos..col_dest]).nil?
-      else
-        col_pos -= 1
+        return true unless game.pieces.find_by(row_position: row_num, col_position: col_pos).nil?
+      end
+    # row and col both decrease
+    elsif row_pos > row_dest && col_pos > col_dest
+      row_dest += 1
+      (row_dest...row_pos).each do |row_num|
         col_dest += 1
-        return !game.pieces.find_by(captured: false, row_position: row_pos, col_position: [col_dest..col_pos]).nil?
+        return true unless game.pieces.find_by(row_position: row_num, col_position: col_dest).nil?
       end
-    elsif (col_pos - col_dest).abs == (row_pos - row_dest).abs # is checking diagonal obstruction
-      # check each row/col combo 1 by 1
-      # row and col both increase
-      if row_pos < row_dest && col_pos < col_dest
-        row_pos += 1
-        (row_pos...row_dest).each do |row_num|
-          col_pos += 1
-          return !game.pieces.find_by(captured: false, row_position: row_num, col_position: col_pos).nil?
-        end
-      # row and col both decrease
-      elsif row_pos > row_dest && col_pos > col_dest
-        row_dest += 1
-        (row_dest...row_pos).each do |row_num|
-          col_dest += 1
-          return !game.pieces.find_by(captured: false, row_position: row_num, col_position: col_dest).nil?
-        end
-      # row increases col decreases
-      elsif row_pos < row_dest && col_pos > col_dest
-        row_pos += 1
-        (row_pos...row_dest).each do |row_num|
-          col_pos -= 1
-          return !game.pieces.find_by(captured: false, row_position: row_num, col_position: col_pos).nil?
-        end
-      # row decreases col increases
-      else
-        row_dest += 1
-        (row_dest...row_pos).each do |row_num|
-          col_dest -= 1
-          return !game.pieces.find_by(captured: false, row_position: row_num, col_position: col_dest).nil?
-        end
+    # row increases col decreases
+    elsif row_pos < row_dest && col_pos > col_dest
+      row_pos += 1
+      (row_pos...row_dest).each do |row_num|
+        col_pos -= 1
+        return true unless game.pieces.find_by(row_position: row_num, col_position: col_pos).nil?
       end
+    # row decreases col increases
     else
-      return "ERROR! in is_obstructed? method line:83"
+      row_dest += 1
+      (row_dest...row_pos).each do |row_num|
+        col_dest -= 1
+        return true unless game.pieces.find_by(row_position: row_num, col_position: col_dest).nil?
+      end
     end
+    false
+  end
+
+  def obstructed?(row_dest, col_dest)
+    return "Invalid input! Knight can't be obstructed." if type == "Knight"
```

Comment:
Sorry -- I edited my comment from when I had this, but maybe you didn't see it.  I think this line should be changed to `return false if type == "Knight"` instead of returning a string.  All this method needs to do with regards to a knight is just let it do whatever it wants (because it can't be obstructed).